examples/dht_node.rs

```diff
@@ -98,10 +100,13 @@ fn main() {
 
     let writer = rx
         .map_err(|_| Error::new(ErrorKind::Other, "rx error"))
-        .fold(sink, move |sink, packet| {
+        .fold(sink, move |sink, (packet, mut addr)| {
             // println!("send = {:?}", packet.clone());
-            debug!("Send {:?} => {:?}", packet.0, packet.1);
-            sink.send(packet)
+            debug!("Send {:?} => {:?}", packet, addr);
+            if let IpAddr::V4(ip) = addr.ip() {
+                addr = SocketAddr::new(IpAddr::V6(ip.to_ipv6_mapped()), 33445);
+            }
+            sink.send((packet, addr))
```

Comment:
What happens if you just send to ipv4 address? Shouldn't tokio take care of this conversion?

Reply:
Now, dht_node run as a ipv6 mode.
If target addr is ipv4, then convert it's address to ipv6 format.

Reply:
Yes, I got it. But why you do it manually? Will it fail otherwise?

Reply:
As I know, it fails with message "Invalid argument"